etna/transforms/embeddings/models/ts2vec.py

```diff
@@ -0,0 +1,287 @@
+            this param helps to reduce the cost of time and memory.
+        Notes
+        -----
+        In case of long series to reduce memory consumption it is recommended to use max_train_length parameter or manually break the series into smaller subseries.
+        """
+        super().__init__(output_dims=output_dims)
+        self.input_dims = input_dims
+        self.output_dims = output_dims
+        self.hidden_dims = hidden_dims
+        self.depth = depth
+        self.max_train_length = max_train_length
+        self.temporal_unit = temporal_unit
+
+        self.device = device
+        self.lr = lr
+        self.batch_size = batch_size
+
+        # Train params
+        self.n_epochs = n_epochs
+        self.n_iters = n_iters
+        self.verbose = verbose
+
+        self.embedding_model = TS2Vec(
+            input_dims=self.input_dims,
+            output_dims=self.output_dims,
+            hidden_dims=self.hidden_dims,
+            depth=self.depth,
+            max_train_length=self.max_train_length,
+            temporal_unit=self.temporal_unit,
+            device=self.device,
+            lr=self.lr,
+            batch_size=self.batch_size,
+        )
+
+        self._is_fitted: bool = False
+
+    def _prepare_data(self, df: pd.DataFrame) -> np.ndarray:
+        """Convert data to array with shapes (n_segments, n_timestamps, input_dims)."""
+        n_timestamps = len(df.index)
+        n_segments = df.columns.get_level_values("segment").nunique()
+        df = df.sort_index(axis=1)
+        x = df.values.reshape((n_timestamps, n_segments, self.input_dims)).transpose(1, 0, 2)
+        return x
+
+    def fit(self, df: pd.DataFrame) -> "TS2VecEmbeddingModel":
+        """Fit the embedding model."""
+        x = self._prepare_data(df=df)
+        self.embedding_model.fit(train_data=x, n_epochs=self.n_epochs, n_iters=self.n_iters, verbose=self.verbose)
+        self._is_fitted = True
+        return self
+
+    def encode_segment(
+        self,
+        df: pd.DataFrame,
+        mask: Union[
+            Literal["binomial"], Literal["continuous"], Literal["all_true"], Literal["all_false"], Literal["mask_last"]
+        ] = "all_true",
+        sliding_length: Optional[int] = None,
+        sliding_padding: int = 0,
+    ) -> np.ndarray:
+        """Create embeddings of the whole series.
+
+        Parameters
+        ----------
+        df:
+            Dataframe with data.
+        mask:
+            The mask used by encoder on the test phase can be specified with this parameter. The possible options are:
+                * 'binomial' - mask timestamp with probability 0.5 (default one, used in the paper). It is used on the training phase.
+                * 'continuous' - mask random windows of timestamps
+                * 'all_true' - mask none of the timestamps
+                * 'all_false' - mask all timestamps
+                * 'mask_last' - mask last timestamp
+        sliding_length:
```

Comment:
Do we need this parameter here?

Reply:
Yes, we generate embeddings for each window (depends on sliding params) and then do maxpooling. Therefore, these parameters are not useless here